mitodl/bootcamp-ecommerce
# Implemented consistent error and success behavior

#### What are the relevant tickets?
Closes #658 
Closes #633 
Closes #919 

#### What's this PR do?
- Adds UI artifacts for instances where certain requests succeed and fail
- Re-enables HTML in site-wide notifications

#### How should this be manually tested?

##### Payment success/failure messages
- Create some order for an in-progress application in django admin. Set the status to "created" and note the id
- Apply this patch
    ```
    diff --git a/main/views.py b/main/views.py
    index afbf03f..51cf424 100644
    --- a/main/views.py
    +++ b/main/views.py
    @@ -60,6 +60,13 @@ def react(request, **kwargs):
                     "purchase_date_utc": request.POST["signed_date_time"],
                 }
             )
    +    context["CSOURCE_PAYLOAD"] = json.dumps(
    +        {
    +            "decision": "ACCEPT",
    +            "bootcamp_run_purchased": "<YOUR BOOTCAMP RUN TITLE>",
    +            "purchase_date_utc": "2021-07-08T00:00:00Z",
    +        }
    +    )
         return render(request, "bootcamp/react.html", context)
 
    ```
- Visit your application dashboard with some qs params: `/applications/?status=receipt&order=<your_order_id>`. You should see a notification thanking you for submitting a payment 
- At this point the app should be polling our API until the order status goes to "fulfilled" or "failed", or the request limit is reached.
    - If you set the status to "fulfilled" within the allotted time, the notification should be the same as the bottom screenshot
    - If you set the status to "failed" within the allotted time, the notification should be the same as the second-to-last screenshot
    - If you let the time elapse without changing the status, the notification should be the same as the second-to-last screenshot

NOTE: These notifications should only come up if (a) those querystring params exist and point to an actual order, and (b) the `purchase_date_utc` value in the context is a maximum of 15 min old.

##### Request failures

- Test the new application form
    - Edit `applications.views.BootcampApplicationViewset.create` to immediately raise an exception or return an error response
    - Use the new application form and click "Continue". You should see an error message
- Test the resume/LinkedIn form
    - Edit `UploadResumeView` to immediately raise an exception or return an error response
    - Using an application with no resume, try to add/edit a LinkedIn url and click "submit". You should see an error message

#### Any background context you want to provide?

- There are some notes in the issue about what to do when certain requests are successful. What I found while I was working on this was that we didn't need them so much because there was already some user feedback. When a new application is created, the drawer closes and the user can see that application immediately. When profile info is edited, the form closes and shows the read-only profile info UI. When a resume/LinkedIn URL are added, the drawer closes and the user's application state will be updated right away in the dashboard. Adding a notification on top of those bits of feedback seems unnecessary for now

#### Screenshots (if appropriate)
<img width="939" alt="ss 2020-07-08 at 17 15 33 " src="https://user-images.githubusercontent.com/14932219/86995164-4835ab80-c176-11ea-94e5-1cbb0161616c.png">
<img width="1320" alt="ss 2020-07-08 at 17 16 32 " src="https://user-images.githubusercontent.com/14932219/86995162-479d1500-c176-11ea-9d3c-68a0f402fb5f.png">
<img width="1294" alt="ss 2020-07-08 at 17 17 29 " src="https://user-images.githubusercontent.com/14932219/86995157-44098e00-c176-11ea-83c5-e7e0253fa0bf.png">
